cpp/include/ucxx/utils/callback_notifier.h

```diff
@@ -2,79 +2,59 @@
  * SPDX-FileCopyrightText: Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.
  * SPDX-License-Identifier: BSD-3-Clause
  */
+#include <atomic>
 #include <condition_variable>
-#include <functional>
-#include <memory>
 #include <mutex>
-#include <utility>
-
 namespace ucxx {
 
 namespace utils {
 
-template <typename Flag>
 class CallbackNotifier {
  private:
-  Flag _flag{};                                  //< flag storing state
+  std::atomic_bool _spinlock{};                  //< spinlock if we're not using condition variables
+  bool _flag{};                                  //< flag storing state
   std::mutex _mutex{};                           //< lock to guard accesses
   std::condition_variable _conditionVariable{};  //< notification condition var
+  bool _use_spinlock{};                          //< should we use the spinlock?
+  bool use_spinlock();
```

Comment:
```suggestion
  bool _useSpinlock{};                          //< should we use the spinlock?
  bool isSpinlockWorkaroundRequired();
```

Reply:
The method and attribute having almost the same name is quite confusing. I renamed both to make a better distinction.